KickLib/Clients/AuthenticationService.cs

```diff
@@ -45,6 +45,7 @@ public async Task AuthenticateAsync(string username, string password)
         payloadPrep.Add("isMobileRequest", true);
         payloadPrep.Add("email", username);
         payloadPrep.Add("password", password);
+        payloadPrep.Add("one_time_password", totp);
```

Comment:
Just one comment, I would add `if` statement here, to add `one_time_password` only of `totp` is not null or empty

Reply:
Yes.  If we add the check to allow an empty totp and adjust the sequence of checks on the loginResponse to detect "2fa_required" before looking for "token" you can attempt to authenticate without a totp.  If the provided credentials are valid and the account does not have authenticator 2FA enabled, the system will respond by sending an email with a code that can be used in a subsequent login attempt.

deb1c7dbdcbaf256e31643d37a20a6424174f30f